projects/plugins/boost/app/modules/optimizations/page-cache/pre-wordpress/boost-cache-actions.php

```diff
@@ -62,7 +63,7 @@ function jetpack_boost_delete_cache_for_home() {
  */
 function jetpack_boost_delete_cache_for_url( $url ) {
 	$boost_cache = new Boost_Cache();
-	$boost_cache->invalidate_cache_for_url( $url, Filesystem_Utils::DELETE_ALL );
+	$boost_cache->delete_page( $url );
```

Comment:
Does this make sense to be `delete_recursive`? I am unsure, asking based on old implementation.

Reply:
Yes, it should be recursive. Before it was modified recently, in version 3.12.1 it did a DELETE_ALL by default. Also, it would allow you to delete or rebuild specific cache files, but I doubt anyone used that functionality.

This function is used by the `jetpack_boost_clear_page_cache_url` hook to allow developers to delete pages. However, [the example given](https://github.com/Automattic/jetpack/blob/update/boost/fix-deleting-directories/projects/plugins/boost/docs/Cache_API.md#delete-the-cache-for-a-specific-url) uses https://example.com/ so a recursive delete would clear the entire cache. I need to update those docs to show a post or page, and not the homepage.

Reply:
I updated the docs and changed this to `delete_recursive`.